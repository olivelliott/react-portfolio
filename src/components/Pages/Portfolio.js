import React from "react";
import { Box, Badge, Image, Flex, Link, chakra } from "@chakra-ui/react";
import ez from "../../assets/projects/ez-stream.png";
import budget from "../../assets/projects/budget-tracker.png";
import insomnia from "../../assets/projects/insomnia.png";
import social from "../../assets/projects/social-network.png";
import weather from "../../assets/projects/weather-dashboard.png";
import technicality from "../../assets/projects/technicality.png";
import hmu from '../../assets/projects/hmu.png';
import { Helmet } from "react-helmet";

function Portfolio() {
  const projectArr = [
    {
      name: 'Hand-Me-Up',
      description: 'A clothing re-sale application with a purpose that reduces the stigma of accepting low-cost clothing. Sellers list their clothing for sale at a rate that just covers shipping. Buyers can shop from a selection of low-cost but trendy and modern clothing. Buyers have the option to make an additional donation to a partner charity of their choice when making a purchase.',
      deployedApp: "https://polar-chamber-11415.herokuapp.com/",
      GitHub: "https://github.com/olivelliott/hand-me-up",
      tech: "React | MongoDB | Apollo | ChakraUI",
      imagePath: hmu
    },
    {
      name: "Ez-Stream",
      description:
        "An application that shows the streaming services a tv show or movie is available on",
      deployedApp: "https://mlmcgeenc.github.io/group-2-project-one/",
      GitHub: "https://github.com/olivelliott/ez-stream",
      tech: "JS | Server Side API",
      imagePath: ez,
    },
    {
      name: "Tech|nicality",
      description:
        "Tech blog site where developers can publish their blog posts and comment on other developers posts.",
      deployedApp: "https://technicality.herokuapp.com/",
      GitHub: "https://github.com/olivelliott/tech-nicality",
      tech: "Node | MySql2 | Express",
      imagePath: technicality,
    },
    {
      name: "Band Site",
      description:
        "A website that features a local band to allow fans to connect with one another and sign up for exclusive updates.",
      deployedApp: "https://band-site-22.herokuapp.com/",
      GitHub: "https://github.com/olivelliott/project-2-band-site",
      tech: "Node | Sequelize | GreenSock",
      imagePath: insomnia,
    },
    {
      name: "Budget Tracker",
      description:
        "A budget tracker application that has been updated to allow for offline access and functionality so that the user can track withdrawals and deposits with or without a data/internet connection.",
      deployedApp: "https://budget-tracker-19-app.herokuapp.com/",
      GitHub: "https://github.com/olivelliott/budget-tracker",
      tech: "PWA",
      imagePath: budget,
    },
    {
      name: "Social Network Api",
      description:
        "An API for a social network web application where users can share their thoughts, react to friends thoughts, and create a friend list.",
      deployedApp: "https://github.com/olivelliott/social-network-api",
      GitHub: "https://github.com/olivelliott/social-network-api",
      tech: "Node | NoSQL | Jest",
      imagePath: social,
    },
    {
      name: "Weather Dashboard",
      description:
        "An application that looks up the current weather conditions based on the location entered.",
      deployedApp: "https://olivelliott.github.io/weather-dashboard/",
      GitHub: "https://github.com/olivelliott/weather-dashboard",
      tech: "JQuery | Server Side API",
      imagePath: weather,
    },
  ];

  return (
    <Flex
      bg="navy"
      _dark={{
        bg: "#3e3e3e",
      }}
      p={50}
      w="full"
      alignItems="center"
      justifyContent="center"
      flexWrap="wrap"
    >
      <Helmet>
        <title>Portfolio</title>
      </Helmet>

      {projectArr.map(
        ({ name, deployedApp, description, GitHub, tech, imagePath }) => {
          return (
            <Box
              m={5}
              bg="gray.800"
              _dark={{
                bg: "gray.800",
              }}
              mx={{
                lg: 8,
              }}
              display={{
                lg: "flex",
              }}
              maxW={{
                lg: "5xl",
              }}
              shadow={{
                lg: "lg",
              }}
              rounded={{
                lg: "lg",
              }}
            >
              <Box w={{ lg: "50%" }}>
                <Image
                  h={{ base: 64, lg: "full", md: "full", sm: "full" }}
                  rounded={{
                    lg: "lg",
                  }}
                  bgSize="cover"
                  src={imagePath}
                ></Image>
              </Box>

              <Box
                py={12}
                px={6}
                maxW={{
                  base: "xl",
                  lg: "5xl",
                }}
                w={{
                  lg: "50%",
                }}
              >
                <chakra.h2
                  fontSize={{
                    base: "2xl",
                    md: "3xl",
                  }}
                  color="white"
                  _dark={{
                    color: "white",
                  }}
                  fontWeight="bold"
                >
                  {name}
                </chakra.h2>
                <Badge
                  borderRadius="full"
                  px="2"
                  backgroundColor="orange"
                  color="gray.800"
                >
                  {tech}
                </Badge>
                <chakra.p
                  mt={4}
                  color="white"
                  _dark={{
                    color: "gray.400",
                  }}
                >
                  {description}
                </chakra.p>

                <Box mt={8}>
                  <Link
                    as="a"
                    target="_blank"
                    href={GitHub}
                    bg="orange"
                    color="gray.100"
                    px={5}
                    py={3}
                    fontWeight="semibold"
                    rounded="lg"
                    _hover={{
                      bg: "white",
                      color: "orange",
                      textDecoration: "none",
                    }}
                  >
                    GitHub
                  </Link>
                </Box>

                <Box mt={8}>
                  <Link
                    as="a"
                    target="_blank"
                    href={deployedApp}
                    bg="orange"
                    color="gray.100"
                    px={5}
                    py={3}
                    fontWeight="semibold"
                    rounded="lg"
                    _hover={{
                      bg: "white",
                      color: "orange",
                      textDecoration: "none",
                    }}
                  >
                    Deployed App
                  </Link>
                </Box>
              </Box>
            </Box>
          );
        }
      )}
    </Flex>
  );
}

export default Portfolio;
