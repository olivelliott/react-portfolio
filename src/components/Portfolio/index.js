import React from "react";
import photo from "../../assets/ez-stream.png";

function Portfolio() {
  const projects = [
    {
      name: "Ez-Stream",
      description:
        "An application that shows the streaming services a tv show or movie is available on",
    },
    {
      name: "Tech|nicality",
      description:
        "Tech blog site where developers can publish their blog posts and comment on other developers posts.",
    },
    {
      name: "Band Site",
      description:
        "A website that features a local band to allow fans to connect with one another and sign up for exclusive updates.",
    },
    {
      name: "Budget Tracker",
      description:
        "A budget tracker application that has been updated to allow for offline access and functionality so that the user can track withdrawals and deposits with or without a data/internet connection.",
    },
    {
      name: "Social Network Api",
      description:
        "An API for a social network web application where users can share their thoughts, react to friends thoughts, and create a friend list.",
    },
    {
      name: "Employee Tracker",
      description:
        "An application that allows a business owner to organize and keep track of the departments, roles, and employees in their company.",
    },
  ];

  function projectSelected() {
    console.log("hi");
  }

  return (
    <section>
      {projects.map((project) => (
        <div className="flex-row">
          <span onClick={projectSelected}>
            <h1 className="mx-1" key={project.name}>
              {project.name}
            </h1>
            <img
              src={photo}
              alt="Ez-Stream project example"
              className="img-thumbnail mx-1"
            />
          </span>
        </div>
      ))}
    </section>
  );
}

export default Portfolio;

// import { Tabs, TabList, TabPanels, Tab, TabPanel } from '@chakra-ui/react'

// function Portfolio() {
//     return (
//         <Tabs>
//             <TabList>
//             <Tab>One</Tab>
//             <Tab>Two</Tab>
//             <Tab>Three</Tab>
//         </TabList>

//         <TabPanels>
//             <TabPanel>
//                 {/* <p>one!</p> */}
//                 <About/>
//             </TabPanel>
//             <TabPanel>
//                 <p>two!</p>
//             </TabPanel>
//             <TabPanel>
//                 <p>three!</p>
//             </TabPanel>
//             </TabPanels>
//         </Tabs>
//     )
// }

// https://chakra-ui.com/docs/components/tabs/usage
