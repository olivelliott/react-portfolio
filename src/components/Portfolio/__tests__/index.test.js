import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Portfolio from '..';

const project =[ 
    { name: 'ez-stream', description: 'web application' }
]

afterEach(cleanup);

describe('Portfolio component', () => {
    it('renders without crashing', () => {
        render(<Portfolio />);
    });

    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Portfolio />);
        expect(asFragment()).toMatchSnapshot();
    })
});

