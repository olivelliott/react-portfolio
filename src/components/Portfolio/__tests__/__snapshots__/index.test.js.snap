// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section>
    <div
      class="flex-row"
    >
      <span>
        <h1
          class="mx-1"
        >
          Ez-Stream
        </h1>
        <img
          alt="Ez-Stream project example"
          class="img-thumbnail mx-1"
          src="ez-stream.png"
        />
      </span>
    </div>
    <div
      class="flex-row"
    >
      <span>
        <h1
          class="mx-1"
        >
          Tech|nicality
        </h1>
        <img
          alt="Ez-Stream project example"
          class="img-thumbnail mx-1"
          src="ez-stream.png"
        />
      </span>
    </div>
    <div
      class="flex-row"
    >
      <span>
        <h1
          class="mx-1"
        >
          Band Site
        </h1>
        <img
          alt="Ez-Stream project example"
          class="img-thumbnail mx-1"
          src="ez-stream.png"
        />
      </span>
    </div>
    <div
      class="flex-row"
    >
      <span>
        <h1
          class="mx-1"
        >
          Budget Tracker
        </h1>
        <img
          alt="Ez-Stream project example"
          class="img-thumbnail mx-1"
          src="ez-stream.png"
        />
      </span>
    </div>
    <div
      class="flex-row"
    >
      <span>
        <h1
          class="mx-1"
        >
          Social Network Api
        </h1>
        <img
          alt="Ez-Stream project example"
          class="img-thumbnail mx-1"
          src="ez-stream.png"
        />
      </span>
    </div>
    <div
      class="flex-row"
    >
      <span>
        <h1
          class="mx-1"
        >
          Employee Tracker
        </h1>
        <img
          alt="Ez-Stream project example"
          class="img-thumbnail mx-1"
          src="ez-stream.png"
        />
      </span>
    </div>
  </section>
</DocumentFragment>
`;
